app/Models/User.php

```diff
@@ -57,4 +62,28 @@ public function roles(): BelongsToMany
     {
         return $this->belongsToMany(Role::class);
     }
+
+    public function fullName(): Attribute
+    {
+        return Attribute::make(
+            get: fn ($value) => $this->attributes['first_name'] . ' ' . $this->attributes['last_name'],
+        );
+    }
+
+    public function username(): Attribute
+    {
+        return Attribute::make(
+            set: fn ($value) => Str::slug($value),
+        );
+    }
+
+    protected static function booted(): void
+    {
+        static::addGlobalScope(new UserActiveScope);
+    }
+
+    public function scopeAdmin(Builder $query): void
+    {
+        $query->where('is_admin', 1);
```

Comment:
```suggestion
        $query->where('is_admin', IS_ADMIN);
```
Đặt const IS_ADMIN = 1 ra chứ để số 1 đọc code k hiểu em nhé

Reply:
ready